include/examples/simple_behavior.h

```diff
@@ -0,0 +1,192 @@
+ * @author Ethan J. Musser (emusser@seas.upenn.edu)
+ * @brief Defines a simple behavior class which commands torques to all joints
+ *        in a SimpleRobot
+ * @date 7/26/22
+ * 
+ * @copyright Copyright 2022 The Trustees of the University of Pennsylvania. All
+ *            rights reserved.
+ * 
+ */
+
+#pragma once
+
+#include <memory>  // std::unique_ptr
+#include <iostream>  // std::cout, std::endl
+#include <numeric>  // std::accumulate
+#include <vector>  // std::vector
+#include "kodlab_mjbots_sdk/behavior.h"  // kodlab::Behavior
+#include "examples/simple_robot.h"  // SimpleRobot
+
+/**
+ * @brief Simple behavior that commands torques to all joints in a SimpleRobot
+ */
+class SimpleBehavior : virtual public kodlab::Behavior<SimpleRobot>
+{
+
+private:
+  /**
+   * @brief The torque to be commanded while motors are running fully
+   */
+  const float RUN_TORQUE = 0;
+
+  /**
+   * @brief The torque to be commanded while motors are being slowed
+   */
+  const float SLOW_TORQUE = 0;
+
+  /**
+   * @brief Enumeration of valid states while behavior is active
+   */
+  enum States { RUNNING, STOPPING } state_;
+
+  /**
+   * @brief The initial joint positions when the behavior begins
+   */
+  std::vector<float> joint_start_positions_;
+
+  /**
+   * @brief The joint velocities
+   */
+  std::vector<float> joint_velocities_;
+
+  /**
+   * @brief The desired joint torques
+   */
+  std::vector<float> desired_torques_;
+
+public:
+
+  /**
+   * @brief Using default Behavior constructor
+   */
+  using Behavior::Behavior;
+
+  /**
+   * @brief Initializes the behavior
+   * @details This is where initialization code, which runs when a behavior is
+   *          first added to a behavior manager, should be put. This is where
+   *          initialization that did not take place in the constructor should
+   *          occur, e.g., setting up pointers. In this example, the desired
+   *          torques are initialized to zero.
+   */
+  void Init() override
+  {
+    // Initializes desired torques (for user safety, these are set to zero)
+    desired_torques_ = std::vector<float>(robot_->joints.size(), 0);
+
+    // Indicate that behavior has been initialized
+    set_initialized();
+    std::fprintf(stdout, "Initialized %s\n", get_name().c_str());
+  }
+
+  /**
+   * @brief Starts the behavior
+   * @details This code runs a single time when the behavior starts up. This is
+   *          a good place to read in the initial robot state and set the
+   *          appropriate variables to begin the behavior. The previously active
+   *          behavior is passed as an argument, permitting different
+   *          functionalities for different prior behaviors. In this example,
+   *          the behavior is marked as active and the initial robot state is
+   *          read.
+   * @param prev_behavior the previously active behavior
+   */
+  void Begin(std::unique_ptr<kodlab::Behavior<SimpleRobot>> &prev_behavior) override
```

Comment:
Don't use a reference to a unique ptr. Why not use a reference to the data stored in the ptr?

Reply:
Switched in d5a5f80f72d7c6a0b4102fe4306e32c69a184078.